packages/beacon-node/src/chain/blocks/blockInput/utils.ts

```diff
@@ -42,3 +47,29 @@ export function convertNewToOldBlobSource(source: BlockSource): BlobsSource {
       return BlobsSource.gossip;
   }
 }
+
+export function emitDataColumnSidecar(
+  emitter: ChainEventEmitter,
+  blockInput: BlockInput | NullBlockInput,
+  blockRoot: Uint8Array
+): void {
+  if (blockInput.block === null) return;
+  if (blockInput.type !== BlockInputType.availableData) return;
+  if (emitter.listenerCount(routes.events.EventType.dataColumnSidecar) === 0) return;
+
+  // TODO: Ideally it would be checked with ForkSeq > fulu but it's not returning right type
+  if (blockInput.blockData.fork !== ForkName.fulu) return;
```

Comment:
this can be something like `if (isForkPreFulu(...)) return;`

Reply:
`if (!isForkPostFulu(blockInput.blockData.fork)) return;` doesn't seem to infer the type correctly but strictly checking for fulu is really error prone and will break in the next fork